fusillade/api/__init__.py

```diff
@@ -0,0 +1,207 @@
+        self.swagger_spec_path = swagger_spec_path
+        self.connexion_app = self.create_connexion_app()
+        self.connexion_full_dispatch_request = self.connexion_app.app.full_dispatch_request
+        self.connexion_request_context = self.connexion_app.app.test_request_context
+        self.trailing_slash_routes = []
+        routes = collections.defaultdict(list)
+        for rule in self.connexion_app.app.url_map.iter_rules():
+            route = re.sub(r"<(.+?)(:.+?)?>", r"{\1}", rule.rule)
+            stripped_route = route.rstrip("/")
+            if route.endswith("/"):
+                self.trailing_slash_routes.append(stripped_route)
+            routes[stripped_route] += rule.methods
+        for route, methods in routes.items():
+            self.route(route, methods=list(set(methods) - {"OPTIONS"}), cors=True)(self.dispatch)
+
+    def create_connexion_app(self):
+        app = FlaskApp('fusillade')
+        # The Flask/Connection app's logger has its own multi-line formatter and configuration. Rather than suppressing
+        # it we let it do its thing, give it a special name and only enable it if Fusillade_DEBUG > 1.
+        # Most of the Fusillade web app's logging is done through the FusilladeChaliceApp.app logger not the Flask
+        # app's logger.
+        app.app.logger_name = 'fus.api'
+        debug = Config.debug_level() > 0
+        app.app.debug = debug
+        app.app.logger.info('Flask debug is %s.', 'enabled' if debug else 'disabled')
+
+        validator_map = {
+            'body': FusilladeRequestBodyValidator,
+            'parameter': FusilladeParameterValidator,
+        }
+
+        resolver = RestyResolver("fusillade.api", collection_endpoint_name="list")
+        app.add_api(self.swagger_spec_path,
+                    validator_map=validator_map,
+                    resolver=resolver,
+                    validate_responses=True,
+                    arguments=os.environ,
+                    options={"swagger_path": self.swagger_spec_path})
+        return app
+
+    def dispatch(self, *args, **kwargs):
+        """
+        This is the main entry point into the connexion application.
+
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        cr = self.current_request
+        context = cr.context
+        uri_params = cr.uri_params or {}
+        method = cr.method
+        query_params = cr.query_params
+        path = context["resourcePath"].format(**uri_params)
+        if context["resourcePath"] in self.trailing_slash_routes:
+            if context["path"].endswith("/"):
+                path += "/"
+            else:
+                return chalice.Response(status_code=requests.codes.found, headers={"Location": path + "/"}, body="")
+        req_body = cr.raw_body if cr._body is not None else None
+        base_url = "https://{}".format(cr.headers["host"]) if cr.headers.get("host") else os.environ["API_DOMAIN_NAME"]
```

Comment:
Please check if client has effective control over the value of host here by way of the host header, and if so, what are the security implications.

Reply:
The users does have control over this value. I will use `API_DOMAIN_NAME`, to prevent the user from abusing this field.